src/validation/schema.js

```diff
@@ -380,15 +380,290 @@ const validateEntityFields = (defs, def) =>
+  return algorithm
+    ? validateFulltextArgumentAlgorithm(def, directive, algorithm)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'algorithm' must be specified`,
+        }),
+      ])
+}
+
+const validateFulltextArgumentAlgorithm = (def, directive, argument) => {
+  if (argument.value.kind != 'EnumValue') {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'algorithm' must be one of: rank, proximityRank`,
+      }),
+    ])
+  } else if (!['rank', 'proximity_rank'].includes(argument.value.value)) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext 'algorithm' value, '${argument.value.value}', must be one of: rank, proximityRank`,
+      }),
+    ])
+  } else {
+    return List([])
+  }
+}
+
+const validateFulltextInclude = (def, directive) => {
+  let include = directive.arguments.find(argument => argument.name.value == 'include')
+  if (include) {
+    if (include.value.kind != 'ListValue') {
+      return List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'include' must be a list`,
+        }),
+      ])
+    }
+    return include.value.values.reduce(
+      (errors, type) =>
+        errors.concat(validateFulltextArgumentInclude(def, directive, type)),
+      List(),
+    )
+  } else {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must be specified`,
+      }),
+    ])
+  }
+}
+
+const validateFulltextArgumentInclude = (def, directive, argument) => {
+  if (argument.kind != 'ObjectValue') {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must have the form '[{ entity: "entityName", fields: [{name: "fieldName"},...]}...]`,
+      }),
+    ])
+  }
+  if (argument.fields.length != 2) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument include must have two fields, 'entity' and 'fields'}`,
+      }),
+    ])
+  }
+  return argument.fields.reduce(
+    (errors, field) =>
+      errors.concat(validateFulltextArgumentIncludeFields(def, directive, field)),
+    List([]),
+  )
+}
+
+const validateFulltextArgumentIncludeFields = (def, directive, field) => {
+  if (!['entity', 'fields'].includes(field.name.value)) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include > ${field.name.value}' must be be one of: [entity, fields]`,
```

Comment:
In other places we aren't using `[]` around the list of possible values (example: `... must be one of: rank, proximityRank`).